src/query/typed_value.cpp

```diff
@@ -362,6 +362,33 @@ bool TypedValue::IsGraph() const { return type_ == Type::Graph; }
 
 #undef DEFINE_VALUE_AND_TYPE_GETTERS
 
+bool TypedValue::ContainsDeleted() const {
+  switch (type_) {
+    case Type::List:
+      for (const auto &elem : list_v) {
+        if (elem.ContainsDeleted()) return true;
+      }
+      return false;
+    case Type::Map:
+      for (const auto &[_, map_value] : map_v) {
+        if (map_value.ContainsDeleted()) return true;
+      }
+      return false;
+    case Type::Vertex:
+      return vertex_v.impl_.vertex_->deleted;
+    case Type::Edge:
+      return edge_v.impl_.edge_.ptr->deleted;
+    case Type::Path:
+      for (const auto &vertex_acc : path_v.vertices()) {
```

Comment:
Can we rewrite this using `any_of` to be consistent?

Reply:
Done